src/mds/MDCache.cc

```diff
@@ -12781,15 +12781,25 @@ int MDCache::dump_cache(std::string_view fn, Formatter *f)
     return 1;
   };
 
+  auto start = mono_clock::now();
+  int64_t count = 0;
   for (auto &p : inode_map) {
     r = dump_func(p.second);
     if (r < 0)
       goto out;
+    if (!(++count % 1000) &&
+	timeout > 0 &&
```

Comment:
`timeout` cannot be 0, why check?

Reply:
MDCache:
    int dump_cache() { return dump_cache({}, nullptr, 0); }
it will set timeout to 0.

Reply:
it is used as a debug function